File: otherLangs/Persona.cpp


Persona::Persona(int id, const std::string& nombre, const std::string& apellido, const std::string& sexo)
    : id(id), nombre(nombre), apellido(apellido), sexo(sexo) {}

int Persona::getID() const {
    return id;
}

void Persona::setID(int id) {
    this->id = id;
}

std::string Persona::getNombre() const {
    return nombre;
}

void Persona::setNombre(const std::string& nombre) {
    this->nombre = nombre;
}

std::string Persona::getApellido() const {
    return apellido;
}

void Persona::setApellido(const std::string& apellido) {
    this->apellido = apellido;
}

std::string Persona::getSexo() const {
    return sexo;
}

void Persona::setSexo(const std::string& sexo) {
    this->sexo = sexo;
}

File: otherLangs/Persona.py
class Persona:
    """
    Datos básicos de una persona, de momento usada por la clase Visitante y EmpleadoEvento
    """
    def __init__(self, id_persona: int, nombre: string, apellido: string, sexo: string):
        self.__id = id_persona
        self.__nombre = nombre
        self.__apellido = apellido
        self.__sexo = sexo

    @property
    def id(self) -> int:
        return self.__id

    @id.setter
    def id(self, value: int):
        self.__id = value

    @property
    def nombre(self) -> string:
        return self.__nombre

    @nombre.setter
    def nombre(self, value: string):
        self.__nombre = value

    @property
    def apellido(self) -> string:
        return self.__apellido

    @apellido.setter
    def apellido(self, value: string):
        self.__apellido = value

    @property
    def sexo(self) -> string:
        return self.__sexo

    @sexo.setter
    def sexo(self, value: string):
        self.__sexo = value